packages/react-native/ReactCommon/react/renderer/core/RawValue.h

```diff
@@ -96,73 +116,142 @@ class RawValue {
    */
   template <typename T>
   explicit operator T() const {
-    return castValue(dynamic_, (T*)nullptr);
+    if (std::holds_alternative<folly::dynamic>(value_)) {
+      folly::dynamic dynamic = std::get<folly::dynamic>(value_);
+      return castValue(dynamic, (T*)nullptr);
+    } else {
+      const auto& [runtime, value] = std::get<JsiPair>(value_);
+      return castValue(value, runtime, (T*)nullptr);
+    }
   }
 
   inline explicit operator folly::dynamic() const noexcept {
-    return dynamic_;
+    return std::get<folly::dynamic>(value_);
   }
 
   /*
    * Checks if the stored value has specified type.
    */
   template <typename T>
   bool hasType() const noexcept {
-    return checkValueType(dynamic_, (T*)nullptr);
+    if (std::holds_alternative<folly::dynamic>(value_)) {
+      folly::dynamic dynamic = std::get<folly::dynamic>(value_);
+      return checkValueType(dynamic, (T*)nullptr);
+    } else {
+      const auto& [runtime, value] = std::get<JsiPair>(value_);
+      return checkValueType(value, runtime, (T*)nullptr);
+    }
   }
 
   /*
    * Checks if the stored value is *not* `null`.
    */
   bool hasValue() const noexcept {
-    return !dynamic_.isNull();
+    if (std::holds_alternative<folly::dynamic>(value_)) {
+      folly::dynamic dynamic = std::get<folly::dynamic>(value_);
+      return !dynamic.isNull();
+    } else {
+      const auto& [runtime, value] = std::get<JsiPair>(value_);
+      return !value.isNull() && !value.isUndefined();
```

Comment:
`std::get<T>(...)`, `jsi::Value::isNull()` and `jsi::Value::isUndefined()` are all **not** `noexcept`, so `hasValue()` can't be `noexcept` anymore.